src/Compilers/CSharp/Portable/Parser/Blender.Cursor.cs

```diff
@@ -55,28 +55,44 @@ private static bool IsNonZeroWidthOrIsEndOfFile(SyntaxNodeOrToken token)
                 return token.Kind() == SyntaxKind.EndOfFileToken || token.FullWidth != 0;
             }
 
-            public Cursor MoveToNextSibling()
+            public static Cursor MoveToNextSibling(Cursor cursor)
```

Comment:
well this diff is just awful.  I recommend side by side diff.  There's now a new top level MoveToNextSibling utility.  It *iteratively* moves the cursor up it's parent chain (instead of using recursion for that).  It then calls into a helper which just checks that particular cursor, without moving it.